code/ship/ship.cpp

```diff
@@ -7402,7 +7402,6 @@ void ship_destroy_instantly(object *ship_objp, int shipnum)
 void ship_cleanup(int shipnum, int cleanup_mode)
 {
 	Assert(shipnum >= 0 && shipnum < MAX_SHIPS);
-	Assert(cleanup_mode & (SHIP_DESTROYED | SHIP_DEPARTED | SHIP_VANISHED));
```

Comment:
I hit this assert when I tested my changes with a red-alert mission so I fixed it since the function can handle other values of `cleanup_mode` properly. I didn't want to put it into another PR since it would have made testing harder but I guess I could do that...